src/Reflection/ReflectionParameter.php

```diff
@@ -497,6 +500,31 @@ private function getClassFromNamedType(ReflectionNamedType $namedType): ?Reflect
         }
     }
 
+    private function detectIsOptional(): bool
+    {
+        if ($this->node->variadic) {
+            return true;
+        }
+
+        if ($this->node->default === null) {
+            return false;
+        }
+
+        foreach ($this->function->getAst()->getParams() as $otherParameterIndex => $otherParameterNode) {
+            if ($otherParameterIndex <= $this->parameterIndex) {
```

Comment:
Is this already covered by our existing tests, perhaps? 🤔

Reply:
Yes, it is: https://github.com/Roave/BetterReflection/blob/5.0.x/test/unit/Reflection/ReflectionParameterTest.php#L488